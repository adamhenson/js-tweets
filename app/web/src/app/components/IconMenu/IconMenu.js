import React from 'react';
import { IconMenu as RTIconMenu } from 'react-toolbox/lib/menu';
import style from './IconMenu.css';

const IconMenu = ({
  children,
}) => (
  <RTIconMenu
    className={style.iconMenu}
    icon="more_horiz"
    position="topRight"
    menuRipple
  >
    { children }
  </RTIconMenu>
);

export default IconMenu;
