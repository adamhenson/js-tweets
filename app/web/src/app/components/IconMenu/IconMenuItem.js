import React from 'react';
import PropTypes from 'prop-types';
import { MenuItem as RTIconMenuItem } from 'react-toolbox/lib/menu';

const IconMenuItem = ({
  caption,
  icon,
  onClick,
  value,
}) => (
  <RTIconMenuItem
    caption={caption}
    icon={icon}
    onClick={onClick}
    value={value}
  />
);

IconMenuItem.propTypes = {
  caption: PropTypes.string.isRequired,
  icon: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired,
};

export default IconMenuItem;
