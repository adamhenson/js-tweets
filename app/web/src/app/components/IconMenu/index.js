export { default as IconMenu } from './IconMenu';
export { default as IconMenuItem } from './IconMenuItem';
