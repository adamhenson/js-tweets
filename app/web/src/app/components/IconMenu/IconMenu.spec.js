import React from 'react';
import IconMenu from './IconMenu';
import IconMenuItem from './IconMenuItem';

test('Render an IconMenu and on click ensure it renders correctly', () => {
  const wrapper = mount(
    <IconMenu>
      <IconMenuItem
        key="hello"
        icon="refresh"
        onClick={() => null}
        value="hello"
        caption="hello"
      />
      <IconMenuItem
        key="world"
        icon="refresh"
        onClick={() => null}
        value="world"
        caption="world"
      />
    </IconMenu>
  );
  expect(wrapper).toMatchSnapshot();
  wrapper.find('[data-react-toolbox="button"]').simulate('click');
  expect(wrapper).toMatchSnapshot();
});
