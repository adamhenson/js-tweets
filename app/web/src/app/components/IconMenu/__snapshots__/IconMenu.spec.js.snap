// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render an IconMenu and on click ensure it renders correctly 1`] = `
<IconMenu>
  <ThemedIconMenu
    className="iconMenu"
    composeTheme="deeply"
    icon="more_horiz"
    iconRipple={true}
    mapThemrProps={[Function]}
    menuRipple={true}
    position="topRight"
    selectable={false}
  >
    <IconMenu
      className="iconMenu"
      icon="more_horiz"
      iconRipple={true}
      menuRipple={true}
      position="topRight"
      selectable={false}
      theme={Object {}}
    >
      <div
        className="iconMenu"
      >
        <ThemedThemed
          composeTheme="deeply"
          disabled={false}
          icon="more_horiz"
          mapThemrProps={[Function]}
          onClick={[Function]}
          ripple={true}
          rippleCentered={true}
          rippleClassName=""
          rippleMultiple={true}
          rippleSpread={2}
        >
          <ThemedRippledComponent
            composeTheme="deeply"
            disabled={false}
            icon="more_horiz"
            mapThemrProps={[Function]}
            onClick={[Function]}
            ripple={true}
            rippleCentered={true}
            rippleClassName=""
            rippleMultiple={true}
            rippleSpread={2}
            theme={Object {}}
          >
            <RippledComponent
              disabled={false}
              icon="more_horiz"
              onClick={[Function]}
              ripple={true}
              rippleCentered={true}
              rippleClassName=""
              rippleMultiple={true}
              rippleSpread={2}
              theme={Object {}}
            >
              <IconButton
                accent={false}
                className=""
                disabled={false}
                icon="more_horiz"
                neutral={true}
                onClick={[Function]}
                onMouseDown={[Function]}
                onTouchStart={[Function]}
                primary={false}
                theme={Object {}}
                type="button"
              >
                <button
                  className=""
                  data-react-toolbox="button"
                  disabled={false}
                  onClick={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchStart={[Function]}
                  type="button"
                >
                  <FontIcon
                    alt=""
                    className=""
                    value="more_horiz"
                  >
                    <span
                      aria-label=""
                      className="material-icons"
                      data-react-toolbox="font-icon"
                    >
                      more_horiz
                    </span>
                  </FontIcon>
                </button>
              </IconButton>
            </RippledComponent>
          </ThemedRippledComponent>
        </ThemedThemed>
        <ThemedMenu
          active={false}
          composeTheme="deeply"
          mapThemrProps={[Function]}
          onHide={[Function]}
          outline={true}
          position="topRight"
          ripple={true}
          selectable={false}
          theme={Object {}}
        >
          <Menu
            active={false}
            onHide={[Function]}
            outline={true}
            position="topRight"
            ripple={true}
            selectable={false}
            theme={Object {}}
          >
            <div
              className=""
              data-react-toolbox="menu"
              style={
                Object {
                  "height": undefined,
                  "width": undefined,
                }
              }
            >
              <div
                style={
                  Object {
                    "height": undefined,
                    "width": undefined,
                  }
                }
              />
              <ul>
                <IconMenuItem
                  caption="hello"
                  icon="refresh"
                  key=".$hello"
                  onClick={[Function]}
                  value="hello"
                >
                  <ThemedThemed
                    caption="hello"
                    composeTheme="deeply"
                    disabled={false}
                    icon="refresh"
                    mapThemrProps={[Function]}
                    onClick={[Function]}
                    ripple={true}
                    rippleCentered={false}
                    rippleClassName=""
                    rippleMultiple={true}
                    rippleSpread={2}
                    value="hello"
                  >
                    <ThemedRippledComponent
                      caption="hello"
                      composeTheme="deeply"
                      disabled={false}
                      icon="refresh"
                      mapThemrProps={[Function]}
                      onClick={[Function]}
                      ripple={true}
                      rippleCentered={false}
                      rippleClassName=""
                      rippleMultiple={true}
                      rippleSpread={2}
                      theme={Object {}}
                      value="hello"
                    >
                      <RippledComponent
                        caption="hello"
                        disabled={false}
                        icon="refresh"
                        onClick={[Function]}
                        ripple={true}
                        rippleCentered={false}
                        rippleClassName=""
                        rippleMultiple={true}
                        rippleSpread={2}
                        theme={Object {}}
                        value="hello"
                      >
                        <MenuItem
                          caption="hello"
                          className=""
                          disabled={false}
                          icon="refresh"
                          onClick={[Function]}
                          onMouseDown={[Function]}
                          onTouchStart={[Function]}
                          selected={false}
                          theme={Object {}}
                          value="hello"
                        >
                          <li
                            className=""
                            data-react-toolbox="menu-item"
                            onClick={[Function]}
                            onMouseDown={[Function]}
                            onTouchStart={[Function]}
                            value="hello"
                          >
                            <FontIcon
                              alt=""
                              className=""
                              value="refresh"
                            >
                              <span
                                aria-label=""
                                className="material-icons"
                                data-react-toolbox="font-icon"
                              >
                                refresh
                              </span>
                            </FontIcon>
                            <span>
                              hello
                            </span>
                          </li>
                        </MenuItem>
                      </RippledComponent>
                    </ThemedRippledComponent>
                  </ThemedThemed>
                </IconMenuItem>
                <IconMenuItem
                  caption="world"
                  icon="refresh"
                  key=".$world"
                  onClick={[Function]}
                  value="world"
                >
                  <ThemedThemed
                    caption="world"
                    composeTheme="deeply"
                    disabled={false}
                    icon="refresh"
                    mapThemrProps={[Function]}
                    onClick={[Function]}
                    ripple={true}
                    rippleCentered={false}
                    rippleClassName=""
                    rippleMultiple={true}
                    rippleSpread={2}
                    value="world"
                  >
                    <ThemedRippledComponent
                      caption="world"
                      composeTheme="deeply"
                      disabled={false}
                      icon="refresh"
                      mapThemrProps={[Function]}
                      onClick={[Function]}
                      ripple={true}
                      rippleCentered={false}
                      rippleClassName=""
                      rippleMultiple={true}
                      rippleSpread={2}
                      theme={Object {}}
                      value="world"
                    >
                      <RippledComponent
                        caption="world"
                        disabled={false}
                        icon="refresh"
                        onClick={[Function]}
                        ripple={true}
                        rippleCentered={false}
                        rippleClassName=""
                        rippleMultiple={true}
                        rippleSpread={2}
                        theme={Object {}}
                        value="world"
                      >
                        <MenuItem
                          caption="world"
                          className=""
                          disabled={false}
                          icon="refresh"
                          onClick={[Function]}
                          onMouseDown={[Function]}
                          onTouchStart={[Function]}
                          selected={false}
                          theme={Object {}}
                          value="world"
                        >
                          <li
                            className=""
                            data-react-toolbox="menu-item"
                            onClick={[Function]}
                            onMouseDown={[Function]}
                            onTouchStart={[Function]}
                            value="world"
                          >
                            <FontIcon
                              alt=""
                              className=""
                              value="refresh"
                            >
                              <span
                                aria-label=""
                                className="material-icons"
                                data-react-toolbox="font-icon"
                              >
                                refresh
                              </span>
                            </FontIcon>
                            <span>
                              world
                            </span>
                          </li>
                        </MenuItem>
                      </RippledComponent>
                    </ThemedRippledComponent>
                  </ThemedThemed>
                </IconMenuItem>
              </ul>
            </div>
          </Menu>
        </ThemedMenu>
      </div>
    </IconMenu>
  </ThemedIconMenu>
</IconMenu>
`;

exports[`Render an IconMenu and on click ensure it renders correctly 2`] = `
<IconMenu>
  <ThemedIconMenu
    className="iconMenu"
    composeTheme="deeply"
    icon="more_horiz"
    iconRipple={true}
    mapThemrProps={[Function]}
    menuRipple={true}
    position="topRight"
    selectable={false}
  >
    <IconMenu
      className="iconMenu"
      icon="more_horiz"
      iconRipple={true}
      menuRipple={true}
      position="topRight"
      selectable={false}
      theme={Object {}}
    >
      <div
        className="iconMenu"
      >
        <ThemedThemed
          composeTheme="deeply"
          disabled={false}
          icon="more_horiz"
          mapThemrProps={[Function]}
          onClick={[Function]}
          ripple={true}
          rippleCentered={true}
          rippleClassName=""
          rippleMultiple={true}
          rippleSpread={2}
        >
          <ThemedRippledComponent
            composeTheme="deeply"
            disabled={false}
            icon="more_horiz"
            mapThemrProps={[Function]}
            onClick={[Function]}
            ripple={true}
            rippleCentered={true}
            rippleClassName=""
            rippleMultiple={true}
            rippleSpread={2}
            theme={Object {}}
          >
            <RippledComponent
              disabled={false}
              icon="more_horiz"
              onClick={[Function]}
              ripple={true}
              rippleCentered={true}
              rippleClassName=""
              rippleMultiple={true}
              rippleSpread={2}
              theme={Object {}}
            >
              <IconButton
                accent={false}
                className=""
                disabled={false}
                icon="more_horiz"
                neutral={true}
                onClick={[Function]}
                onMouseDown={[Function]}
                onTouchStart={[Function]}
                primary={false}
                theme={Object {}}
                type="button"
              >
                <button
                  className=""
                  data-react-toolbox="button"
                  disabled={false}
                  onClick={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchStart={[Function]}
                  type="button"
                >
                  <FontIcon
                    alt=""
                    className=""
                    value="more_horiz"
                  >
                    <span
                      aria-label=""
                      className="material-icons"
                      data-react-toolbox="font-icon"
                    >
                      more_horiz
                    </span>
                  </FontIcon>
                </button>
              </IconButton>
            </RippledComponent>
          </ThemedRippledComponent>
        </ThemedThemed>
        <ThemedMenu
          active={true}
          composeTheme="deeply"
          mapThemrProps={[Function]}
          onHide={[Function]}
          outline={true}
          position="topRight"
          ripple={true}
          selectable={false}
          theme={Object {}}
        >
          <Menu
            active={true}
            onHide={[Function]}
            outline={true}
            position="topRight"
            ripple={true}
            selectable={false}
            theme={Object {}}
          >
            <div
              className=""
              data-react-toolbox="menu"
              style={
                Object {
                  "height": 0,
                  "width": 0,
                }
              }
            >
              <div
                style={
                  Object {
                    "height": 0,
                    "width": 0,
                  }
                }
              />
              <ul
                style={
                  Object {
                    "clip": "rect(0 0px 0px 0)",
                  }
                }
              >
                <IconMenuItem
                  caption="hello"
                  icon="refresh"
                  key=".$hello"
                  onClick={[Function]}
                  value="hello"
                >
                  <ThemedThemed
                    caption="hello"
                    composeTheme="deeply"
                    disabled={false}
                    icon="refresh"
                    mapThemrProps={[Function]}
                    onClick={[Function]}
                    ripple={true}
                    rippleCentered={false}
                    rippleClassName=""
                    rippleMultiple={true}
                    rippleSpread={2}
                    value="hello"
                  >
                    <ThemedRippledComponent
                      caption="hello"
                      composeTheme="deeply"
                      disabled={false}
                      icon="refresh"
                      mapThemrProps={[Function]}
                      onClick={[Function]}
                      ripple={true}
                      rippleCentered={false}
                      rippleClassName=""
                      rippleMultiple={true}
                      rippleSpread={2}
                      theme={Object {}}
                      value="hello"
                    >
                      <RippledComponent
                        caption="hello"
                        disabled={false}
                        icon="refresh"
                        onClick={[Function]}
                        ripple={true}
                        rippleCentered={false}
                        rippleClassName=""
                        rippleMultiple={true}
                        rippleSpread={2}
                        theme={Object {}}
                        value="hello"
                      >
                        <MenuItem
                          caption="hello"
                          className=""
                          disabled={false}
                          icon="refresh"
                          onClick={[Function]}
                          onMouseDown={[Function]}
                          onTouchStart={[Function]}
                          selected={false}
                          theme={Object {}}
                          value="hello"
                        >
                          <li
                            className=""
                            data-react-toolbox="menu-item"
                            onClick={[Function]}
                            onMouseDown={[Function]}
                            onTouchStart={[Function]}
                            value="hello"
                          >
                            <FontIcon
                              alt=""
                              className=""
                              value="refresh"
                            >
                              <span
                                aria-label=""
                                className="material-icons"
                                data-react-toolbox="font-icon"
                              >
                                refresh
                              </span>
                            </FontIcon>
                            <span>
                              hello
                            </span>
                          </li>
                        </MenuItem>
                      </RippledComponent>
                    </ThemedRippledComponent>
                  </ThemedThemed>
                </IconMenuItem>
                <IconMenuItem
                  caption="world"
                  icon="refresh"
                  key=".$world"
                  onClick={[Function]}
                  value="world"
                >
                  <ThemedThemed
                    caption="world"
                    composeTheme="deeply"
                    disabled={false}
                    icon="refresh"
                    mapThemrProps={[Function]}
                    onClick={[Function]}
                    ripple={true}
                    rippleCentered={false}
                    rippleClassName=""
                    rippleMultiple={true}
                    rippleSpread={2}
                    value="world"
                  >
                    <ThemedRippledComponent
                      caption="world"
                      composeTheme="deeply"
                      disabled={false}
                      icon="refresh"
                      mapThemrProps={[Function]}
                      onClick={[Function]}
                      ripple={true}
                      rippleCentered={false}
                      rippleClassName=""
                      rippleMultiple={true}
                      rippleSpread={2}
                      theme={Object {}}
                      value="world"
                    >
                      <RippledComponent
                        caption="world"
                        disabled={false}
                        icon="refresh"
                        onClick={[Function]}
                        ripple={true}
                        rippleCentered={false}
                        rippleClassName=""
                        rippleMultiple={true}
                        rippleSpread={2}
                        theme={Object {}}
                        value="world"
                      >
                        <MenuItem
                          caption="world"
                          className=""
                          disabled={false}
                          icon="refresh"
                          onClick={[Function]}
                          onMouseDown={[Function]}
                          onTouchStart={[Function]}
                          selected={false}
                          theme={Object {}}
                          value="world"
                        >
                          <li
                            className=""
                            data-react-toolbox="menu-item"
                            onClick={[Function]}
                            onMouseDown={[Function]}
                            onTouchStart={[Function]}
                            value="world"
                          >
                            <FontIcon
                              alt=""
                              className=""
                              value="refresh"
                            >
                              <span
                                aria-label=""
                                className="material-icons"
                                data-react-toolbox="font-icon"
                              >
                                refresh
                              </span>
                            </FontIcon>
                            <span>
                              world
                            </span>
                          </li>
                        </MenuItem>
                      </RippledComponent>
                    </ThemedRippledComponent>
                  </ThemedThemed>
                </IconMenuItem>
              </ul>
            </div>
          </Menu>
        </ThemedMenu>
      </div>
    </IconMenu>
  </ThemedIconMenu>
</IconMenu>
`;
